packages/wrangler/src/autoconfig/frameworks/next.ts

```diff
@@ -0,0 +1,127 @@
+import { statSync } from "node:fs";
+import { readFile, writeFile } from "node:fs/promises";
+import { brandColor, dim } from "@cloudflare/cli/colors";
+import { spinner } from "@cloudflare/cli/interactive";
+import { dedent } from "../../utils/dedent";
+import { installPackages } from "../c3-vendor/packages";
+import { AutoConfigFrameworkConfigurationError } from "../errors";
+import { appendToGitIgnore } from "../git";
+import { usesTypescript } from "../uses-typescript";
+import { Framework } from ".";
+import type { ConfigurationOptions } from ".";
+import type { RawConfig } from "@cloudflare/workers-utils";
+
+export class NextJs extends Framework {
+	name = "next.js";
+
+	preview = "opennextjs-cloudflare build && opennextjs-cloudflare preview";
+	deploy = "opennextjs-cloudflare build && opennextjs-cloudflare deploy";
+
+	async configure({
+		dryRun,
+		projectPath,
+	}: ConfigurationOptions): Promise<RawConfig> {
+		const usesTs = usesTypescript(projectPath);
+
+		const nextConfigPath = probeForNextConfigPath(usesTs);
+		if (!nextConfigPath) {
+			throw new AutoConfigFrameworkConfigurationError(
+				"No Next.js configuration file could be detected. Note: only next.config.ts and next.config.mjs files are supported."
+			);
+		}
+
+		if (!dryRun) {
+			await installPackages(["@opennextjs/cloudflare@^1.3.0"], {
+				startText: "Installing @opennextjs/cloudflare adapter",
+				doneText: `${brandColor("installed")}`,
+			});
+
+			await updateNextConfig(nextConfigPath);
+
+			await createOpenNextConfigFile(projectPath);
+
+			await appendToGitIgnore(
+				projectPath,
+				dedent`
+				# OpenNext
+				.open-next
+				`,
+				{
+					startText: "Adding open-next section to .gitignore file",
+					doneText: `${brandColor(`added`)} open-next section to .gitignore file`,
+				}
+			);
+		}
+
+		return {
+			main: ".open-next/worker.js",
+			compatibility_flags: ["nodejs_compat"],
+			assets: {
+				binding: "ASSETS",
+				directory: ".open-next/assets",
+			},
+		};
+	}
+
+	configurationDescription = "Configuring project for Next.js with OpenNext";
+}
+
+function probeForNextConfigPath(usesTs: boolean): string | undefined {
+	const pathsToProbe = [
+		...(usesTs ? ["next.config.ts"] : []),
+		"next.config.mjs",
+	];
+
+	for (const path of pathsToProbe) {
+		const stats = statSync(`next.config.${usesTs ? "ts" : "mjs"}`, {
```

Comment:
use `path`instead ?

Reply:
yes... I refactored this bit and I must have forgotten to use the `statSync` argument 🤦 

thanks! 👍